### PR TITLE
Use antsibull 0.40.1 for docsite build

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Use the [latest release](https://github.com/ansible-community/antsibull/releases/tag/0.40.0) of antsibull to build the docsite. It uses RST tables instead of HTML blobs for option and return value tables, and these tables are responsive.

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
docsite

### CODE REVIEW

**Review by webknjaz — APPROVED** [review]

**Comment by felixfontein** [comment]
0.40.0 crashed during the test docsite build; 0.40.1 fixes the crash (antsibull now has tests to prevent this kind of crash).

**Review by samccann — APPROVED** [review]
verified on jenkins and local build that this works.
fwiw I changed max_threads to 1 on my vm to get the build to work w/o timeouts.

**Review by samccann — CHANGES_REQUESTED** [review]
Forgot one more place this needs to change I think. We need to keep the docs_build requirements file in sync with this. 
https://github.com/ansible/ansible/blob/devel/test/sanity/code-smell/docs-build.requirements.txt#L7

**Comment by samccann** [comment]
Hmm top of that file (docs-build.requirements.txt sez to generate it w/ another file so might need edits somewhere else? or maybe just regenerate the file??

**Comment by felixfontein** [comment]
@samccann I think changing that file prevents this from being backported to any stable branch.

**Comment by felixfontein** [comment]
On the other hand, the code-smell test is internal to ansible/ansible, so maybe this doesn't block backporting.

**Comment by samccann** [comment]
I don't know if we want to backport this or not really. I know I'd like to keep this on devel only 'for a time' so we can get any feedback from users. I can put it in the bullhorn and on reddit etc once it's on devel. I was thinking it would go live on /latest/ with Ansible 6?

**Comment by samccann** [comment]
...or are you thinking of updating that docs-build.requirements in a separate PR? I'm still not sure we'd want our docs requirements out of sync with the sanity test...

**Comment by felixfontein** [comment]
I've now updated it in this PR. I'm not sure whether this is a problem, but now the dependencies for the docs build differ partially from the dependencies for the changelog test (https://github.com/ansible/ansible/blob/devel/test/lib/ansible_test/_data/requirements/sanity.changelog.txt). @mattclay is this a problem (and do the requirements have to match), or is it ok if they differ?

**Comment by felixfontein** [comment]
It looks to me like all test deps should be updated at once, or that the files need to be edited manually (despite the boilerplate warning). Anyway, the Sphinx theme should be updated as well, 0.7.0 is pretty old and some bugs have been fixed since then.

**Comment by mattclay** [comment]
Manually update the pinned requirements in `test/sanity/code-smell/docs-build.requirements.txt` that changed.

It should be safe to do so as long as those package versions are compatible with the other pinned packages in that file.

Once this is merged I'll rebuild the test container used for the `devel` branch.

If it does get backported, there will be merge conflicts and I'll need to rebuild the container used in the `stable-2.12` branch. I'd like to avoid that if we can, but if it's needed then that's OK.

File: lib/ansible/plugins/doc_fragments/default_callback.py [hunk 1]
```diff
@@ -18,9 +18,9 @@ class ModuleDocFragment(object):
         env:
           - name: DISPLAY_SKIPPED_HOSTS
             deprecated:
-              why: environment variables without "ANSIBLE_" prefix are deprecated
+              why: environment variables without C(ANSIBLE_) prefix are deprecated
```
Comment (mattclay) [inline lib/ansible/plugins/doc_fragments/default_callback.py]: Should the quote fixes be done in a separate PR so they can be independently backported?

Comment (mattclay) [reply lib/ansible/plugins/doc_fragments/default_callback.py]: Ah, I see you've already opened a PR for that.

Comment (felixfontein) [reply lib/ansible/plugins/doc_fragments/default_callback.py]: Sorry, didn't mention it here, but I created a separate PR (#76698). I've now rebased after that one was merged.

**Review by mattclay — APPROVED** [review]

**Review by samccann — APPROVED** [review]

**Comment by felixfontein** [comment]
@webknjaz @samccann @mattclay thanks for reviewing and testing!
